policy/predicate/file.go

```diff
@@ -184,16 +255,36 @@ func (pred *ModifiedLines) Evaluate(ctx context.Context, prctx pull.Context) (bo
 		deletions += int64(f.Deletions)
 	}
 
-	for expr, v := range map[ComparisonExpr]int64{
-		pred.Additions: additions,
-		pred.Deletions: deletions,
-		pred.Total:     additions + deletions,
-	} {
-		if !expr.IsEmpty() && expr.Evaluate(v) {
-			return true, "", nil
+	if !pred.Additions.IsEmpty() {
+		predicateResult.Values = []string{fmt.Sprintf("+%d", additions)}
+		predicateResult.ConditionValues = []string{"added lines" + pred.Additions.String()}
+		if pred.Additions.Evaluate(additions) {
+			predicateResult.Satisfied = true
+			return &predicateResult, nil
+		}
+	}
```

Comment:
I'd structure this as a single `switch` statement, which is a common way to write if-else changes in Go:


```go
switch {
case !pred.Additions.Empty() && pred.Additions.Evaluate(additions):
  predicateResult.Satisfied = true
  predicateResult.Values = []string{fmt.Sprintf("+%d lines", additions)}
  predicateResult.ConditionValues = []string{"added lines " + pred.Additions.String()}
  return &predicateResult, nil

case !pred.Deletions.Empty() && pred.Deletions.Evaluate(deletions):
  predicateResult.Statisfied = true
  // ...

// additional cases here
}
```

Reply:
I don't think I can put it in a `switch` statement actually. Because I am accumulating the `Values` and `ConditionValues` in each if condition. So in the case the PR does not satisfy the predicate, The `Values` and `ConditionValues` would contain information about all the three cases if not empty. If it satisfy the predicate, then it would only return the relevant condition. I would not be able to accumulate values using a `switch` statement.

Reply:
Sorry, you're right, I missed the accumulation that was happening.